astropy/timeseries/downsample.py

```diff
@@ -61,45 +81,93 @@ def aggregate_downsample(time_series, *, time_bin_size=None, time_bin_start=None
     if not isinstance(time_series, TimeSeries):
         raise TypeError("time_series should be a TimeSeries")
 
-    if not isinstance(time_bin_size, u.Quantity):
-        raise TypeError("time_bin_size should be a astropy.unit quantity")
+    if time_bin_size is not None and not isinstance(time_bin_size, (u.Quantity, TimeDelta)):
+        raise TypeError("'time_bin_size' should be a Quantity or a TimeDelta")
 
-    bin_size_sec = time_bin_size.to_value(u.s)
+    if time_bin_start is not None and not isinstance(time_bin_start, (Time, TimeDelta)):
+        time_bin_start = Time(time_bin_start)
+
+    if time_bin_end is not None and not isinstance(time_bin_end, (Time, TimeDelta)):
+        time_bin_end = Time(time_bin_end)
 
     # Use the table sorted by time
     sorted = time_series.iloc[:]
```

Comment:
Good catch! I was concerned about this too when I was making the changes, but decided to keep it as is since it was part of the original code. Will change :)